From 32a5d025aa40d4aea3b7b9f3626fc5d2771f8033 Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Mon, 25 Jun 2012 18:20:23 +0800
Subject: [PATCH 296/486] ENGR00176977-1 reconstruct audio hal

1. judge the sound card in the hal,not distinct by the property
2. support wm8958, wm8962, hdmi

Signed-off-by: b02247 <b02247@freescale.com>
---
 alsa/imx_audio_hal.cpp | 16 ++++++----------
 1 file changed, 6 insertions(+), 10 deletions(-)

diff --git a/alsa/imx_audio_hal.cpp b/alsa/imx_audio_hal.cpp
index 0e9ea58..272a827 100644
--- a/alsa/imx_audio_hal.cpp
+++ b/alsa/imx_audio_hal.cpp
@@ -49,16 +49,12 @@ static int imx_adev_open(const hw_module_t* module, const char* name,
     bool found = false;
 
     /*find the audio hal for different board*/
-    property_get("ro.product.device", value, "");
-
-    if(!strcmp(value, "sabresd_6q") || !strcmp(value, "sabresd_6dl") || !strcmp(value, "imx6sl_arm2")) {
-        ret = hw_get_module(AUDIO_HARDWARE_MODULE_ID_TINYALSA, &module_audio);
-        if(ret)   found = false;
-        else {
-            ret = module_audio->methods->open(module, AUDIO_HARDWARE_INTERFACE,(struct hw_device_t**)device);
-            if(ret)  found = false;
-            else     found = true;
-        }
+    ret = hw_get_module(AUDIO_HARDWARE_MODULE_ID_TINYALSA, &module_audio);
+    if(ret)   found = false;
+    else {
+        ret = module_audio->methods->open(module, AUDIO_HARDWARE_INTERFACE,(struct hw_device_t**)device);
+        if(ret)  found = false;
+        else     found = true;
     }
 
     if(!found) {
-- 
1.8.0

